### DISCUSSION TITLE
PUSH FTXUI THROUGH GLFW?

### DISCUSSION THREAD

**Discussion by mightgoyardstill** [ArthurSonzogni/FTXUI#534] [Ideas]
Hey, wondering if its possible to push the rendering from FTXUI through to a windowing library such as glfw?

**Comment by ArthurSonzogni** [comment]
That's not the intended use of FTXUI, but you can likely do it via:
1. Use a terminal emulator library, plug FTXUI as a client, by redirecting std::cin and std::cout.
2. Inspect the ftxui terminal, and render it yourself. Plug manually the events for from the GFLW events.

**Reply by mightgoyardstill** [reply]
hi thanks for the reply on this. i was wondering if you had any working examples of being able to redirect std::cin and std::cout from ftxui?

does the Screen or terminal classes hold an internal buffer which is flushed?

**Reply by ArthurSonzogni** [reply]
I saw @zedshaw who streamed on Twich writting a pixel backend for FTXUI:
https://x.com/lzsthw/status/1839033134790357352


About redirecting `std::cout` to a string buffer. Here is an example:
```cpp
#include <iostream>
#include <sstream>

int main() {
    std::stringstream ss;
    
    // Save the original buffer of std::cout
    std::streambuf* oldCoutStreamBuf = std::cout.rdbuf();
    
    // Redirect std::cout to stringstream
    std::cout.rdbuf(ss.rdbuf());
    
    // Anything you print to cout will go to ss now
    std::cout << "Hello, world!" << std::endl;
    
    // Restore original std::cout buffer
    std::cout.rdbuf(oldCoutStreamBuf);
    
    // Now you can use the stringstream content
    std::cout << "Captured stringstream content: " << ss.str() << std::endl;
    
    return 0;
}
```

Note that you can directly use FTXUI `Screen` buffer directly to read the cells:

```cpp
Screen::PixelAt(x,y)
```
